### PR TITLE
Rewrite SuperInterfaces to use Elements

### PR DESCRIPTION
**Pull request by alexvanyo**
This PR has two commits:

- The first adds failing tests for #14 , #15 , and #22.
- The second fixes those failing tests and those, through a rewrite of `SuperInterfaces.kt`. The initial implementation made too many assumptions, and in particular went straight to a Kotlin-only representation via `TypeSpec`. This broke Java interoperability, and also prevented analyzing the visibility of interfaces. The new implementation can traverse both Java and Kotlin base classes and interfaces, and supports mixes of both to flesh out edge cases.

### CODE REVIEW

File: sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/SuperInterfaces.kt [hunk 1]
```diff
@@ -11,35 +11,82 @@ import com.squareup.kotlinpoet.TypeName
...
+    typeElement: TypeElement,
+    typeSpec: TypeSpec
+): List<TypeName> {
     val superInterfaces = mutableListOf<TypeName>()
-    getAllSuperInterfacesImpl(
-        typeSpec,
-        typeSpec.wildcardedTypeVariables,
-        superInterfaces
+
+    getAllSuperInterfacesKotlinImpl(
```
[the remark this answers is not in the document]
Comment (alexvanyo) [reply sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/SuperInterfaces.kt:49]: Yep, I'm taking advantage of the fact that our entry point (the `sealed class` we're making a `sealed enum` for) has to be a Kotlin class.

File: sealedenumprocessor/src/test/kotlin/com/livefront/sealedenum/JavaBaseClasses.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
+package com.livefront.sealedenum
+
+import org.intellij.lang.annotations.Language
+
+interface JavaBaseClassesInterface1
```
[the remark this answers is not in the document]
Comment (alexvanyo) [reply sealedenumprocessor/src/test/kotlin/com/livefront/sealedenum/JavaBaseClasses.kt:5]: Agreed, which is why I kinda just kept adding interfaces to cover more edge cases. I'll make them named more explicitly!

[the remark this answers is not in the document]
Comment (alexvanyo) [reply sealedenumprocessor/src/test/kotlin/com/livefront/sealedenum/JavaBaseClasses.kt]: Alright, the main reason why names were getting annoying was because of avoiding conflicts with all of the other tests. In fa5ba7e I bit the bullet and moved all of the compilation tests into loosely related subpackages.

This allowed me to rename these interfaces in the newly named `kitchensink` tests to simply `KotlinInterface*` without fear of running into that same name elsewhere in 285bdbe.

Sort of unrelated, but I added b0f522e to make `getSourceFile` work for both `.kt` and `.java` files as well.

**Review by cdflynn — APPROVED** [review]
Nice!  If you get a chance (doesn't have to be this PR) we should add `local.properties` to the `.gitignore`.

**Comment by alexvanyo** [comment]
Thanks @cdflynn !
